src/OpenTelemetry.Instrumentation.Wcf/Implementation/ActionMetadata.cs

```diff
@@ -18,7 +18,13 @@ namespace OpenTelemetry.Instrumentation.Wcf;
 
 internal sealed class ActionMetadata
 {
-    public string ContractName { get; set; }
+    public ActionMetadata(string? contractName, string operationName)
+    {
```

Comment:
It might be nice to have a `Debug.Assert(operationName != null, "operationName was null")` here. The annotations I think are great but easy to abuse like someone _could_ still do `new ActionMetadata(null, null!)` if they wanted to.